pkg/pci/pci.go

```diff
@@ -46,3 +48,74 @@ func (p *PCI) ReadConfig() error {
 	p.ExtraInfo = append(p.ExtraInfo, hex.Dump(c))
 	return nil
 }
+
+type barreg struct {
+	offset int64
+	*os.File
+}
+
+func (r *barreg) Read(b []byte) (int, error) {
+	return r.ReadAt(b, r.offset)
+}
+
+func (w *barreg) Write(b []byte) (int, error) {
+	return w.WriteAt(b, w.offset)
+}
+
+// ReadConfigRegister reads a configuration register of size 8, 16, 32, or 64.
+// It will only work on little-endian machines.
+func (p *PCI) ReadConfigRegister(offset, size int64) (uint64, error) {
+	f, err := os.Open(filepath.Join(p.FullPath, "config"))
+	if err != nil {
+		return 0, err
+	}
+	defer f.Close()
+	var reg uint64
+	r := &barreg{offset: offset, File: f}
+	switch size {
+	default:
+		return 0, fmt.Errorf("%d is not valid: only options are 8, 16, 32, 64")
+	case 64:
+		err = binary.Read(r, binary.LittleEndian, &reg)
+	case 32:
+		var val uint32
+		err = binary.Read(r, binary.LittleEndian, &val)
+		reg = uint64(val)
+	case 16:
+		var val uint16
+		err = binary.Read(r, binary.LittleEndian, &val)
+		reg = uint64(val)
+	case 8:
+		var val uint8
+		err = binary.Read(r, binary.LittleEndian, &val)
+		reg = uint64(val)
+	}
+	return reg, err
+}
+
+// WriteConfigRegister writes a configuration register of size 8, 16, 32, or 64.
+// It will only work on little-endian machines.
+func (p *PCI) WriteConfigRegister(offset, size int64, val uint64) error {
+	f, err := os.OpenFile(filepath.Join(p.FullPath, "config"), os.O_WRONLY, 0)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	w := &barreg{offset: offset, File: f}
+	switch size {
+	default:
+		return fmt.Errorf("%d is not valid: only options are 8, 16, 32, 64")
```

Comment:
`pkg/pci/pci.go:107: missing argument for Errorf("%d"): format reads arg 1, have only 0 args`

Reply:
done